Sources/PackageLoading/ManifestLoader+Validation.swift

```diff
@@ -78,10 +78,14 @@ public struct ManifestValidator {
                 }
             }
 
-            // Check that products that reference only binary targets don't define a type.
-            let areTargetsBinary = product.targets.allSatisfy { self.manifest.targetMap[$0]?.type == .binary }
-            if areTargetsBinary && product.type != .library(.automatic) {
-                diagnostics.append(.invalidBinaryProductType(productName: product.name))
+            // Check that products that reference only binary targets don't define an explicit library type.
+            if product.targets.allSatisfy({ self.manifest.targetMap[$0]?.type == .binary }) {
+                switch product.type {
+                case .library(.automatic), .executable:
+                    break
+                default:
+                    diagnostics.append(.invalidBinaryProductType(productName: product.name))
+                }
```

Comment:
If I understand this correctly, this still allows products that have both a binary target and a source target to work right? Do we want to to allow this at all? What happens if a I have a library product that depends on a source target and a binaryTarget that is an executable?

Reply:
This is the validation for the executable product itself which can only vend a singular target which now can either be an executable source target or a binary target that contains an executable.